cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/replicationtool/hbase11/TableUpdater.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.replicationtool.hbase11;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.replication.StatusToolUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hbase.CoprocessorEnvironment;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.apache.hadoop.hbase.client.HTableInterface;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+
+/**
+ * Common functionality required by Replication State Coprocessors to hold updates in memory and
+ * flush into HBase periodically.
+ */
+public class TableUpdater {
+
+  private static final Log LOG = LogFactory.getLog(TableUpdater.class);
+  private String columnFamily;
+  private String clusterID;
+  private UUID rsID;
+  HTableInterface hTableInterface;
+  CoprocessorEnvironment env;
+  StatusToolUtils utils = new StatusToolUtils();
+
+  public TableUpdater (String clusterID, String columnFamily, CoprocessorEnvironment env) {
+    this.clusterID = clusterID;
+    this.columnFamily = columnFamily;
```

Comment:
Changed this to be columnQualifier which is provided by the two Coprocessors when instantiating the class.